app/src/routes/project/LayersPanel/index.tsx

```diff
@@ -4,10 +4,61 @@ import { EditorMode } from '@/lib/models';
 import { PinLeftIcon, PinRightIcon } from '@radix-ui/react-icons';
 import clsx from 'clsx';
 import { observer } from 'mobx-react-lite';
-import { useState } from 'react';
+import { useCallback, useState } from 'react';
 import { useEditorEngine } from '..';
 import LayersTab from './LayersTab';
 import { capitalizeFirstLetter } from '/common/helpers';
+import { ReactComponentDescriptor } from '/electron/main/code/components';
+import { MainChannels } from '/common/constants';
+import { Button } from '@/components/ui/button';
+
+function ScanComponentsButton() {
+    const editorEngine = useEditorEngine();
+
+    const onClick = useCallback(async () => {
```

Comment:
Note: We're planning to save the project's folder at setup. @drfarrell already has some design drafted up so we won't have to search for the folder at this point. This works in the meantime